services/src/main/java/org/keycloak/services/managers/ResourceAdminManager.java

```diff
@@ -241,15 +242,18 @@ protected Response sendBackChannelLogoutRequestToClientUri(ClientModel resource,
 
     public GlobalRequestResult logoutAll(RealmModel realm) {
         realm.setNotBefore(Time.currentTime());
-        Stream<ClientModel> resources = realm.getClientsStream()
-          .filter(c -> { try { c.getClientId(); return true; } catch (Exception ex) { return false; } } );
 
         GlobalRequestResult finalResult = new GlobalRequestResult();
         AtomicInteger counter = new AtomicInteger(0);
-        resources.forEach(r -> {
-            counter.getAndIncrement();
-            GlobalRequestResult currentResult = logoutClient(realm, r, realm.getNotBefore());
-            finalResult.addAll(currentResult);
+        realm.getClientsStream().forEach(c -> {
+            try {
+                counter.getAndIncrement();
+                GlobalRequestResult currentResult = logoutClient(realm, c, realm.getNotBefore());
+                finalResult.addAll(currentResult);
+            } catch (ModelIllegalStateException ex) {
+                // currently, GlobalRequestResult doesn't allow for information about clients that we were unable to retrieve.
+                logger.warn("unable to retrieve model information, skipping entity", ex);
```

Comment:
Message again: now that we are in the context of a logout there's no mention of it in the log message. Perhaps it is unnecessary but to me the different messages are confusing.

Reply:
Thanks for spotting this. Apparently I mixed up the two log message you commented on, the next version will have them switched.